File: acestep_studio/debug_supabase.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');

const envPath = path.join(__dirname, '.env.local');
console.log("Reading:", envPath);

try {
    const content = fs.readFileSync(envPath, 'utf-8');
    const env = {};
    content.split(/\r?\n/).forEach(line => {
        line = line.trim();
        if (!line || line.startsWith('#')) return;
        const idx = line.indexOf('=');
        if (idx !== -1) {
            const key = line.substring(0, idx).trim();
            let val = line.substring(idx + 1).trim();
            if ((val.startsWith('"') && val.endsWith('"')) || (val.startsWith("'") && val.endsWith("'"))) {
                val = val.slice(1, -1);
            }
            env[key] = val;
        }
    });

    const rootEnvPath = path.join(__dirname, '..', '.env');
    console.log("Reading Root:", rootEnvPath);
    if (fs.existsSync(rootEnvPath)) {
        const rootContent = fs.readFileSync(rootEnvPath, 'utf-8');
        rootContent.split(/\r?\n/).forEach(line => {
            line = line.trim();
            if (!line || line.startsWith('#')) return;
            const idx = line.indexOf('=');
            if (idx !== -1) {
                const key = line.substring(0, idx).trim();
                let val = line.substring(idx + 1).trim();
                if ((val.startsWith('"') && val.endsWith('"')) || (val.startsWith("'") && val.endsWith("'"))) {
                    val = val.slice(1, -1);
                }
                // Don't overwrite if exists (prefer local?) Actually root usually has secrets.
                // Prioritize root for SERVICE KEY
                if (key === 'SUPABASE_SERVICE_ROLE_KEY' || !env[key]) {
                    env[key] = val;
                }
            }
        });
    }

    const url = env.NEXT_PUBLIC_SUPABASE_URL;
    const key = env.SUPABASE_SERVICE_ROLE_KEY || env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

    console.log("URL:", url ? "Found" : "Missing");
    console.log("Key Type:", key ? (env.SUPABASE_SERVICE_ROLE_KEY ? "Service" : "Anon") : "Missing");

    if (!url || !key) {
        console.error("Credentials missing from .env.local");
        process.exit(1);
    }

    const supabase = createClient(url, key);

    async function run() {
        console.log("Listing Buckets...");
        const { data: buckets, error } = await supabase.storage.listBuckets();
        if (error) {
            console.error("List Error:", error);
            if (error.status === 401 || error.status === 403) {
                console.log("Permission Denied. Anon key cannot list buckets. Need Service Role.");
            }
        } else {
            const names = buckets.map(b => b.name);
            console.log("Buckets:", names);
            if (!names.includes('music')) {
                console.log("Creating 'music' bucket...");
                const { error: createError } = await supabase.storage.createBucket('music', { public: true });
                if (createError) console.error("Create Error:", createError);
                else console.log("Bucket 'music' Created Successfully!");
            } else {
                console.log("Bucket 'music' exists.");
            }
        }
    }
    run();

} catch (e) {
    console.error("File Read Error:", e);
}

File: debug_supabase.py
import os
import sys
from dotenv import load_dotenv
from supabase import create_client, Client

# Load env variables from the studio folder
env_path = os.path.join("acestep_studio", ".env.local")
if not os.path.exists(env_path):
    print(f"Error: {env_path} not found")
    sys.exit(1)

load_dotenv(env_path)

url = os.getenv("NEXT_PUBLIC_SUPABASE_URL")
# Prefer Service Role for admin tasks
key = os.getenv("SUPABASE_SERVICE_ROLE_KEY") 
key_type = "SERVICE_ROLE"

if not key:
    key = os.getenv("NEXT_PUBLIC_SUPABASE_ANON_KEY")
    key_type = "ANON"

print(f"URL: {url}")
print(f"Key Type: {key_type}")

if not url or not key:
    print("Error: Missing Supabase credentials")
    sys.exit(1)

try:
    supabase: Client = create_client(url, key)
    
    print("--- Checking Storage Buckets ---")
    buckets = supabase.storage.list_buckets()
    bucket_names = [b.name for b in buckets]
    print(f"Current Buckets: {bucket_names}")
    
    if "music" not in bucket_names:
        print("Bucket 'music' is MISSING. Attempting to create...")
        try:
            supabase.storage.create_bucket("music", options={"public": True})
            print("Successfully created 'music' bucket.")
        except Exception as create_err:
            print(f"Creation Failed: {create_err}")
            print("Hint: If using ANON key, Creation might be blocked by Policies.")
    else:
        print("Bucket 'music' exists.")
        
except Exception as e:
    print(f"Operation Failed: {e}")
